src/tools/clippy/tests/ui/declare_interior_mutable_const/others.rs

```diff
@@ -1,5 +1,5 @@
 #![warn(clippy::declare_interior_mutable_const)]
-
+#![allow(interior_mutable_consts)]
```

Comment:
How do those Clippy lints interact with the new rustc lint?

Reply:
This proposed lint is similar to `clippy::declare_interior_mutable_const`, except that contrary to the Clippy lint which recurse into the type and can lint on arbitrary type that have interior mutability, this lint only checks if the type is marked with `#[rustc_significant_interior_mutable_type]` (which only concerns a few core/std types).

So for these types (and only these types) we would report a similar warning. (Preventing this duplicated warning should be a matter of checking if the type as the attribute).

Reply:
I would prefer adding this check to the Clippy lints, over just allowing the rustc lint in the tests.

The tests could then be adapted to remove all test cases that get linted by `rustc`, except for maybe 1 to check that the Clippy lint doesn't actually trigger on those core/std types.